java/core/src/java/org/apache/orc/impl/ColumnStatisticsImpl.java

```diff
@@ -170,7 +165,198 @@ public int hashCode() {
     }
   }
 
-  private static final class IntegerStatisticsImpl extends ColumnStatisticsImpl
+  /**
+   * Column statistics for List and Map types.
+   */
+  private static final class CollectionColumnStatisticsImpl extends ColumnStatisticsImpl
+      implements CollectionColumnStatistics {
+
+    protected long minimum = Long.MAX_VALUE;
+    protected long maximum = Long.MIN_VALUE;
+    protected long sum = 0;
+    protected boolean hasMinimum = false;
+    protected boolean overflow = false;
+
+    CollectionColumnStatisticsImpl() {
+      super();
+    }
+
+    CollectionColumnStatisticsImpl(OrcProto.ColumnStatistics stats) {
+      super(stats);
+      OrcProto.CollectionStatistics collStat = stats.getCollectionStatistics();
+      if (collStat.hasMinChildren()) {
+        hasMinimum = true;
+        minimum = collStat.getMinChildren();
+      }
+      if (collStat.hasMaxChildren()) {
+        maximum = collStat.getMaxChildren();
+      }
+      if (collStat.hasTotalChildren()) {
+        sum = collStat.getTotalChildren();
+      } else {
+        overflow = true;
+      }
+    }
+
+    @Override
+    public void updateCollectionLength(final long length) {
+      /*
+       * Here, minimum = minCollectionLength
+       * maximum = maxCollectionLength
+       * sum = childCount
+       */
+      if (!hasMinimum) {
+        hasMinimum = true;
+        minimum = length;
+        maximum = length;
+      } else if (length < minimum) {
+        minimum = length;
+      } else if (length > maximum) {
+        maximum = length;
+      }
+
+      this.sum += length;
+    }
+
+    @Override
+    public void reset() {
+      super.reset();
+      hasMinimum = false;
+      minimum = Long.MAX_VALUE;
+      maximum = Long.MIN_VALUE;
+      sum = 0;
+      overflow = false;
+    }
+
+    @Override
+    public void merge(ColumnStatisticsImpl other) {
+      if (other instanceof CollectionColumnStatisticsImpl) {
+        CollectionColumnStatisticsImpl otherColl = (CollectionColumnStatisticsImpl) other;
+        if (!hasMinimum) {
+          hasMinimum = otherColl.hasMinimum;
+          minimum = otherColl.minimum;
+          maximum = otherColl.maximum;
+        } else if (otherColl.hasMinimum) {
+          if (otherColl.minimum < minimum) {
+            minimum = otherColl.minimum;
+          }
+          if (otherColl.maximum > maximum) {
+            maximum = otherColl.maximum;
+          }
+        }
+
+        overflow |= otherColl.overflow;
+        if (!overflow) {
+          boolean wasPositive = sum >= 0;
```

Comment:
the sum for the lengths will always be positive (assuming you can't overflow).

Reply:
agreed, I will simplify this.